File: Factorial.cs
using System;

class Factorial
{
    static void Main()
    {
        // Call the method to compute the factorial
        ComputeFactorial();
    }

    // Method to compute the factorial
    static void ComputeFactorial()
    {
        Console.Write("Enter a positive integer: ");
        int n = Convert.ToInt32(Console.ReadLine());  // Take user input

        // Check if the entered number is a positive integer
        if (n < 0)
        {
            Console.WriteLine("Please enter a positive integer.");
            return;  // Exit the method if the number is not valid
        }

        // Initialize factorial result to 1 (since 0! = 1)
        long factorial = 1;

        // Using a while loop to compute factorial
        int i = 1;
        while (i <= n)
        {
            factorial *= i;  // Multiply the current number with factorial
            i++;  // Increment i
        }

        // Print the result
        Console.WriteLine($"The factorial of {n} is: {factorial}");
    }
}

File: factorial.js
// Read command-line argument n 
let n = parseInt(process.argv[2]); 
 
// Validate input 
if (isNaN(n) || n < 0) { 
    console.log("Please enter a valid non-negative integer."); 
    process.exit(1); 
} 
 
// Initialize factorial as 1 
let factorial = 1; 
 
// Calculate factorial using a for loop 
for (let i = 1; i <= n; i++) { 
    factorial *= i; 
} 
 
// Print result 
console.log(Factorial of ${n} is: ${factorial});